apps/salesforce-commerce-cloud/src/components/dialog/ProductSearchResults.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import { css } from '@emotion/react';
+
+import { Box, Flex, SkeletonContainer, SkeletonImage } from '@contentful/f36-components';
+import { AssetIcon } from '@contentful/f36-icons';
+import tokens from '@contentful/f36-tokens';
+
+import {
+  SearchResultsProps,
+  SearchResultProps,
+  height,
+  headerHeight,
+  stickyHeaderBreakpoint,
+} from './SearchPicker';
+
+const ProductSearchResults = (props: SearchResultsProps) => {
+  const resultsStyles = css`
+    max-height: ${height}px;
+    padding: ${tokens.spacingL} ${tokens.spacingL} 0 ${tokens.spacingL};
+    @media screen and (min-height: ${stickyHeaderBreakpoint}px;) {
+      padding: calc(${tokens.spacingL + headerHeight}px) ${tokens.spacingL} 0 ${tokens.spacingL};
+    }
+  `;
+
+  return (
+    <Box as="section" css={resultsStyles}>
+      <div
+        css={css`
+          display: flex;
+          flex-wrap: wrap;
+          margin-left: -${tokens.spacingS};
+          margin-right: -${tokens.spacingS};
+        `}>
+        {props.searchResults.map((result) => (
+          <ProductSearchResult
+            key={props.fieldType === 'product' ? result.id : `${result.catalogId}:${result.id}`}
+            fieldType={props.fieldType}
+            onItemSelect={props.onItemSelect}
+            selected={props.selectedItems}
+            result={result}
+          />
+        ))}
+      </div>
+    </Box>
+  );
+};
+
+const ProductSearchResult = (props: SearchResultProps) => {
```

Comment:
Same comment as before, I feel this would be nicer to move into it's own file for separation of concerns